### DISCUSSION TITLE
Complex property rule question

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#2660] [Questions]
How do I set the value of a collection row that is determined in the rule?
I have a list of crews that have a list of unit items.  I look up the crew, get the correct crew, then look up the unit item.  If the unit item is found, update it, if not add it to the collection.  This is easy peazy with _crew.CrewUnittemList.Add(upi); 
But with the _context.AddOutValue(ConstructionBidInfoPersist.CrewProperty.UnitItemProperty, upi);  I'm lost.  

Is there an example of how to do this?  I've looked through the blogs from Johnny B and the samples, but didn't come across one that address _dynamically finding grandchild and setting the properties._

**Comment by rockfordlhotka** [comment]
Generally the answer is to use the Target property from the context, which (in a sync rule) allows you to access the entire object graph.

There is no good answer for this in an async rule.
